assembl/static2/js/app/pages/survey.jsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { graphql } from 'react-apollo';
+import gql from 'graphql-tag';
+import { I18n, Translate } from 'react-redux-i18n';
+import { Grid, Button } from 'react-bootstrap';
+import Alert from '../components/common/alert';
+import Modal from '../components/common/modal';
+import Loader from '../components/common/loader';
+import Video from '../components/debate/survey/video';
+import Header from '../components/debate/survey/header';
+import Question from '../components/debate/survey/question';
+import Navigation from '../components/debate/survey/navigation';
+import Proposals from '../components/debate/survey/proposals';
+import { getIfPhaseCompletedByIdentifier } from '../utils/timeline';
+
+class Survey extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      moreProposals: false,
+      showModal: false
+    };
+    this.showMoreProposals = this.showMoreProposals.bind(this);
+    this.getIfProposals = this.getIfProposals.bind(this);
+    this.redirectToLogin = this.redirectToLogin.bind(this);
+    this.displayAlert = this.displayAlert.bind(this);
+  }
+  getIfProposals(questions) {
+    this.questions = questions;
+    if (!this.questions) return false;
+    let isProposals = false;
+    this.questions.forEach((question) => {
+      if (question.posts.edges.length > 0) isProposals = true;
+    });
+    return isProposals;
+  }
+  showMoreProposals() {
+    this.setState({
+      moreProposals: true
+    });
+  }
+  redirectToLogin() {
+    const { connectedUserId } = this.props.context;
+    this.setState({ showModal: !connectedUserId });
```

Comment:
No. This specifications from the PO. If you want to vote or post a proposal, you have to be connected. The PO wants to have the login in a modal. So I created a modal with a button to be redirected to the login view and after merge we will put the login inside the modal.